byron/ledger/executable-spec/src/Ledger/Update/Generators.hs

```diff
@@ -103,41 +113,185 @@ pparamsGen =
 
-listGen :: Word64 -> Word64 -> Int -> Int -> Gen [(Slot, (ProtVer, PParams))]
-listGen loSl hiSl loLen hiLen = Gen.list (Range.linear loLen hiLen) inOneSlot
+listFads :: Word64 -> Word64 -> Int -> Int -> Gen [(Slot, (ProtVer, PParams))]
+listFads loSl hiSl loLen hiLen = Gen.list (Range.linear loLen hiLen) inOneSlot
  where
   inOneSlot :: Gen (Slot, (ProtVer, PParams))
   inOneSlot = (\s v p -> (s, (v, p)))
-    <$> slotGen loSl hiSl
-    <*> protVerGen
-    <*> pparamsGen
+    <$> CG.slot loSl hiSl
+    <*> protVer
+    <*> pparams
 
 -- | Generates an environment for the PVBUMP STS with an empty list of
 -- updates
-pvbumpEmptyListEnvGen :: Gen (Environment PVBUMP)
-pvbumpEmptyListEnvGen = (, []) <$> slotGen 0 100000
+pvbumpEmptyListEnv :: Gen (Environment PVBUMP)
+pvbumpEmptyListEnv = (, []) <$> CG.slot 0 100000
 
 -- | Generates an environment for the PVBUMP STS such that s_n <= 2 *
 -- k
-pvbumpBeginningsEnvGen :: Gen (Environment PVBUMP)
-pvbumpBeginningsEnvGen =
+pvbumpBeginningsEnv :: Gen (Environment PVBUMP)
+pvbumpBeginningsEnv =
   (,)
-    <$> (slotGen 0 $ 2 * (unBlockCount k))
-    <*> listGen 0 100000 0 10
+    <$> (CG.slot 0 $ 2 * (unBlockCount k))
+    <*> listFads 0 100000 0 10
 
 -- | Generates an environment for the PVBUMP STS such that s_n >= 2 *
 -- k
-pvbumpAfter2kEnvGen :: Gen (Environment PVBUMP)
-pvbumpAfter2kEnvGen =
+pvbumpAfter2kEnv :: Gen (Environment PVBUMP)
+pvbumpAfter2kEnv =
   let kv = unBlockCount k in (,)
-    <$> slotGen (2 * kv + 1) (10 * kv)
+    <$> CG.slot (2 * kv + 1) (10 * kv)
     <*> ((++)
-         <$> listGen 0 1 1 1 -- to ensure there is at least one
-                             -- element left after domain restriction
-                             -- in the lastProposal property
-         <*> listGen 0 (10 * kv) 1 10
+         <$> listFads 0 1 1 1 -- to ensure there is at least one element
+                          -- left after domain restriction in the
+                          -- lastProposal property
+         <*> listFads 0 (10 * kv) 1 10
         )
 
 -- | Generates a state value for the PVBUMP STS
-pvbumpStateGen :: Gen (State PVBUMP)
-pvbumpStateGen = (,) <$> protVerGen <*> pparamsGen
+pvbumpState :: Gen (State PVBUMP)
+pvbumpState = (,) <$> protVer <*> pparams
+
+-- | Generates a judgement context for the PVBUMP STS for its property #1
+emptyPVUpdateJC :: Gen (JC PVBUMP)
+emptyPVUpdateJC = (,,)
+  <$> pvbumpEmptyListEnv
+  <*> pvbumpState
+  <*> pure ()
+
+-- | Generates a judgement context for the PVBUMP STS for its property #2
+beginningsNoUpdateJC :: Gen (JC PVBUMP)
+beginningsNoUpdateJC = (,,)
+  <$> pvbumpBeginningsEnv
+  <*> pvbumpState
+  <*> pure ()
+
+-- | Generates a judgement context for the PVBUMP STS for its property #3
+lastProposalJC :: Gen (JC PVBUMP)
+lastProposalJC = (,,)
+  <$> pvbumpAfter2kEnv
+  <*> pvbumpState
+  <*> pure ()
+
+
+-- | Generates an @ApName@
+apName :: Gen ApName
+apName = ApName <$> Gen.element ["byron", "shelley", "praos"]
```

Comment:
Check with the core team (e.g. @erikd) what appropriate application names we might want to use as examples. In `cardano-sl` we use [this](https://github.com/input-output-hk/cardano-sl/blob/aca578c8b9500b8b043e51b395b74b93337d477e/wallet/test/integration/configuration.yaml#L70) for instance.

Reply:
Actual strings don't matter here, do they? `ApName` is just a wrapper around `String` and as far as the STS under test is considered, they can be anything, including a complete Mandarin translation of all Shakespeare's works.

Reply:
Yeah, but I'd check if `byron` is an appropriate app name. Otherwise I rather have a any other string, lest people looking at this repo would think we don't understand what an application name actually is.

Reply:
Sure. Fixing it in the upcoming commit.